lab_3_keywords_textrank/main.py

```diff
@@ -173,7 +192,16 @@ def extract_pairs(tokens: tuple[int, ...], window_length: int) -> Optional[tuple
     In case of corrupt input data, None is returned:
     tokens must not be empty, window lengths must be integer, window lengths cannot be less than 2.
     """
-    pass
+    if not tokens or not type_check(window_length, int) or window_length < 2:
+        return None
+    pairs = []
+    for i in range(len(tokens) + 1 - window_length):
+        window = tokens[i: i + window_length]
+        for token_1 in window:
+            for token_2 in window:
+                if token_1 < token_2 and (token_1, token_2) not in pairs:
```

Comment:
can you explain `token_1 < token_2` ?

Reply:
sure
1) we'd like to avoid pairs with the same tokens, e.g (3, 3)
2) we don't want to have duplicates like (2, 3) and (3, 2)
My algorithm select pairs only with unique tokens in ascending order